client/commonFramework/add-loop-protect.js

```diff
@@ -4,13 +4,16 @@ window.common = (function(global) {
     common = { init: [] }
   } = global;
 
-  loopProtect.hit = function hit(line) {
-     var err = 'Error: Exiting potential infinite loop at line ' +
-     line +
-     '. To disable loop protection, write: \n\\/\\/ noprotect\nas the first' +
-     'line. Beware that if you do have an infinite loop in your code' +
-     'this will crash your browser.';
-    console.error(err);
+  loopProtect.hit = function(line) {
+    var msg = 'Potential infinite loop at line ' +
+      line +
+      '. To disable loop protection, write:' +
+      ' \\n\\/\\/ noprotect\\nas the first' +
+      ' line. Beware that if you do have an infinite loop in your code' +
+      ' this will crash your browser.';
+    window.__err = new Error(msg);
```

Comment:
Cool. Thanks for investigating. Looks like we just need to remove the lines where the error is being attached to the window object and everything else should work fine.